### PR TITLE
Feature-invite guest via email complete

### LINKED ISSUE #50: Ability to invite guests
As a user 
I want the ability to invite my friends
So that i don't eat dinner by myself

A.C.

'Invite' page built based on: [invision](https://projects.invisionapp.com/d/main#/console/10296633/220424117/preview)

### PR DESCRIPTION
**Pull request by justinl-y**
Closes #50 (Outstanding issues up to connection to reducer)

### CODE REVIEW

File: js/components/EmailInviteIndicator/index.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import React, { PropTypes } from 'react';
+import { View, Text } from 'react-native';
+import styles from './styles';
+
+const EmailInviteIndicator = ({ number }) => {
+  return (
+    <View>
+      <Text>{number || 0}<Text> Invites</Text></Text>
```
Comment (BlakeTurner) [inline js/components/EmailInviteIndicator/index.js:8]: Did you mean to nest those elements like that?

File: js/components/InviteForm/index.js [hunk 2]
```diff
@@ -18,69 +20,93 @@ class InviteForm extends Gandalf {
...
+      const validEmailCount = Object.keys(validEmails).length;
+      return validEmailCount;
+    } else {
+      return emailCount;
+    }
+  }
+
+  addEmailField = () => {
+    const fields = this.state.fields;
+    const lastEmail = (fields[Object.keys(fields)[Object.keys(fields).length - 1]]);
```
Comment (BlakeTurner) [inline js/components/InviteForm/index.js]: My brain just broke.
I'd suggest breaking this up a bit:

```js
const lastEmailKey = Object.keys(fields).pop();
const lastEmail = fields[lastEmailKey];
```

File: js/components/InviteForm/index.js [hunk 3]
```diff
@@ -18,69 +20,93 @@ class InviteForm extends Gandalf {
...
+
+    this.setState({ emailCount: this.numberOfValidEmails() });
+
+    if (lastEmail.value !== '' && /.+@.+\..+/.test(lastEmail.value)) {
```
Comment (BlakeTurner) [inline js/components/InviteForm/index.js]: An empty string is falsey, so you could use:

```js
if (lastEmail.value && /.+@.+\..+/.test(lastEmail.value))
```

**Review by BlakeTurner — CHANGES_REQUESTED** [review]
There's some really good work here. The patterns you're developing are great for re-use. Just a few refactoring comments.

File: js/components/EmailInviteIndicator/index.js (hunk 1 above)
Comment (justinl-y) [reply js/components/EmailInviteIndicator/index.js:8]: Yes. The UI needs these text elements to be differently styled.  I added the structure for later style props.

File: js/components/InviteForm/index.js (hunk 2 above)
Comment (justinl-y) [reply js/components/InviteForm/index.js]: Done.

File: js/components/InviteForm/index.js (hunk 3 above)
Comment (justinl-y) [reply js/components/InviteForm/index.js]: Done.

**Comment by justinl-y** [comment]
PR comments actioned.

**Review by BlakeTurner — APPROVED** [review]

**Pull request merged**